src/app/core/services/user.service.ts

```diff
@@ -17,7 +17,18 @@ export class UserService {
     return of(regUsers);
   }
 
-
+  public keepChanges(newUser: any ): Observable<IUser> {
```

Comment:
There are a couple of issues
1) Rename it to updateUser
2) Avoid any type
3) You can rewrite the user fully. Use destructurization, f.e 
`user = {...newData, id: user.id}`
Don't use for cycle